File: rescue-app-backend/Models/Animal.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace rescueApp.Models
{
    public class Animal
    {
        public int Id { get; set; }
        public string? AnimalType { get; set; }
        public string? Name { get; set; }
        public string? Breed { get; set; }
        public DateTime? DateOfBirth { get; set; }
        public string? Gender { get; set; }
        public decimal? Weight { get; set; }
        public string? Story { get; set; }
        public string? AdoptionStatus { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime DateUpdated { get; set; }
        public Guid? CreatedByUserId { get; set; }
        public Guid? UpdatedByUserId { get; set; }

        public virtual ICollection<AdoptionHistory>? AdoptionHistories { get; set; }
        public virtual ICollection<AnimalDocument>? AnimalDocuments { get; set; }
        public virtual ICollection<AnimalImage> AnimalImages { get; set; }

        [ForeignKey("CreatedByUserId")]
        public virtual User? CreatedByUser { get; set; }

        [ForeignKey("UpdatedByUserId")]
        public virtual User? UpdatedByUser { get; set; }

        public Animal()
        {
            AdoptionHistories = new HashSet<AdoptionHistory>();
            AnimalImages = new HashSet<AnimalImage>();
        }
    }
}

File: rescue-app-frontend/src/types/animal.ts
import { AnimalImage } from './animalImage'; // Import the image type

export interface Animal {
	id: number;
	animalType?: string | null;
	name?: string | null;
	species?: string | null;
	breed?: string | null;
	dateOfBirth?: string | null;
	gender?: string | null;
	weight?: number | null;
	story?: string | null;
	adoptionStatus?: string | null;
	dateCreated: string;
	dateUpdated: string;
	createdByUserId?: string | null;
	updatedByUserId?: string | null;
	animalImages?: AnimalImage[] | null;
	currentFosterUserId?: string | null;
	currentFosterName?: string | null;
}
